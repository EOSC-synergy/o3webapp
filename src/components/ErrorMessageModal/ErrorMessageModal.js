import React from "react";
import {useTheme} from "@mui/material/styles";
import {Button, Card, Grid, Modal, Typography} from "@mui/material";
import CancelOutlinedIcon from '@mui/icons-material/CancelOutlined';
import PropTypes from 'prop-types';

/**
 * Displays an error message.
 * @component
 * 
 * @param {Object} props
 * @param {boolean} props.isOpen -> whether the error message modal should be displayed
 * @param {function} props.onClose -> handles closing of the modal
 * @param {String} props.message -> error message
 * @returns {JSX.Element} a jsx file containing a modal with the given error message
 */
function ErrorMessageModal(props) {

    /**
     * The style of the modal.
     *
     * @type {{boxShadow: number, transform: string, top: string, bgColor: *, left: string, minWidth: string, position: string}}
     */
    const style = {
        position: 'absolute',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        bgColor: useTheme().palette.grey[200],
        boxShadow: 24,
    };

    return (
        <Modal
            open={props.isOpen}
            onClose={props.onClose}
        >
            <Card sx={style}>
                <Grid
                    container
                    direction="column"
                >
                    <Grid
                        item
                        key="header"
                        sx={{backgroundColor: "#e85e6c", display: "flex", justifyContent: "center", py: 5}}
                    >
                        <CancelOutlinedIcon color="white" sx={{fontSize: '100px'}}/>
                    </Grid>
                    <Grid
                        item
                        key="message"
                        sx={{px: 5, pt: 5}}>
                        <Typography variant="h3" sx={{textAlign: "center"}}>Ooops!</Typography>
                        <br/>
                        <Typography>{props.message}</Typography>
                    </Grid>
                    <Grid
                        item
                        key="button"
                        sx={{px: 5, pb: 5, display: "flex", justifyContent: "right"}}
                    >
                        <Button onClick={props.onClose}>Dismiss</Button>
                    </Grid>
                </Grid>
            </Card>
        </Modal>
    );
}

ErrorMessageModal.propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired
}

export default ErrorMessageModal;