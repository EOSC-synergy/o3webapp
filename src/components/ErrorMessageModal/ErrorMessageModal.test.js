import React from 'react';
import ErrorMessageModal from './ErrorMessageModal';
import { render } from '@testing-library/react';


describe('test ErrorMessageModal component rendering', () => {
    it('renders without crashing', () => {
        render(<ErrorMessageModal isOpen={true} onClose={jest.fn()} />);
    });

    it('renders as expected', () => {
        const { baseElement } = render(<ErrorMessageModal isOpen={true} onClose={jest.fn()} />);
        expect(baseElement).toMatchSnapshot();
    });
});
