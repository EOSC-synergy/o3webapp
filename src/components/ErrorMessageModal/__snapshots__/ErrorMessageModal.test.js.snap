// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorMessageModal component rendering renders as expected 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root css-79ws1d-MuiModal-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1xb2fui-MuiPaper-root-MuiCard-root"
      tabindex="-1"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-h814oq-MuiGrid-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorWhite MuiSvgIcon-fontSizeMedium css-5jyhtf-MuiSvgIcon-root"
            data-testid="CancelOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.59-13L12 10.59 8.41 7 7 8.41 10.59 12 7 15.59 8.41 17 12 13.41 15.59 17 17 15.59 13.41 12 17 8.41z"
            />
          </svg>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-11ewa74-MuiGrid-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-47suzj-MuiTypography-root"
          >
            Ooops!
          </h3>
          <br />
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-1d9xmeq-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Dismiss
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
